### ISSUE TITLE
Feature Request: Retrieve RStudio's list of dependent packages

### ISSUE THREAD

**Issue by coatless** [rstudio/rstudio#2332]
I would greatly appreciate the ability to retrieve a list of _RStudio_ dependent _R_ packages. This list looks like it is hard-coded in: 

[/src/gwt/src/org/rstudio/studio/client/common/dependencies/DependencyManager.java](https://github.com/rstudio/rstudio/blob/6830fabb7d50025fa3be5e4bcfdace0154b5e6d3/src/gwt/src/org/rstudio/studio/client/common/dependencies/DependencyManager.java)

I'm making this request because some dependencies get out of sync. For example, the _RMarkdown_ dependencies are presently out of date: 

https://github.com/rstudio/rstudio/blob/b2923217abb72df07c7c198482ba6cbd89a1ffe9/src/gwt/src/org/rstudio/studio/client/common/dependencies/DependencyManager.java#L237-L238

**Comment by ronblum** [comment]
Thanks @coatless for the suggestion!

**Comment by jmcphers** [comment]
How do you want to retrieve the list, and what do you want to use the list for?

I've got a PR open at https://github.com/rstudio/rstudio/pull/5585 which will make what you want possible (at least from our end), but am not sure what you're asking for -- an API? documentation?

**Comment by coatless** [comment]
@jmcphers an API that would return something along the lines of: 

```
rstudioapi::rstudio_r_packages()
# [1] "caTools" "bitops" ... 
```

The version may also be nice to have, but we're primarily needing it to know dependencies to install for RStudio to function.

**Comment by jmcphers** [comment]
Okay, just added an API!

**Comment by coatless** [comment]
@jmcphers excellent. So, the call here would be:

```
if(exists("getPackageDependencies")) {
   required_rstudio_pkgs = getPackageDependencies()
   install.packages(required_rstudio_pkgs, repos = "https://cloud.r-project.org")
} 
```

**Comment by jmcphers** [comment]
Yes, close! the function will be called `.rs.api.getPackageDependencies()` and you'll generally want to invoke it through rstudioapi (we will add an accessor on the API side once the RStudio side is merged)

**Comment by coatless** [comment]
Great. Would you like me to create an issue over on the `rstudioapi` repo?

**Comment by jmcphers** [comment]
@coatless This is all working end to end now; with the dev versions of RStudio and rstudioapi, you can get the list from `rstudioapi::getRStudioPackageDependencies()`.
